lte/gateway/c/oai/tasks/mme_app/mme_config.c

```diff
@@ -830,6 +830,64 @@ int mme_config_parse_file(mme_config_t* config_pP) {
         }
       }
     }
+
+    // RESTRICTED PLMN SETTING
+    setting = config_setting_get_member(
+        setting_mme, MME_CONFIG_STRING_RESTRICTED_PLMN_LIST);
+    config_pP->restricted_plmn.num = 0;
+    OAILOG_INFO(LOG_MME_APP, "MME_CONFIG_STRING_RESTRICTED_PLMN_LIST \n");
+    if (setting != NULL) {
+      num = config_setting_length(setting);
+      OAILOG_INFO(
+          LOG_MME_APP, "Number of restricted PLMNs configured =%d\n", num);
+      AssertFatal(
+          num <= MAX_RESTRICTED_PLMN,
+          "Number of restricted PLMNs configured:%d exceeds number of "
+          "restricted PLMNs supported :%d \n",
+          num, MAX_RESTRICTED_PLMN);
+
+      for (i = 0; i < num; i++) {
+        sub2setting = config_setting_get_elem(setting, i);
+
+        if (sub2setting != NULL) {
+          if ((config_setting_lookup_string(
+                  sub2setting, MME_CONFIG_STRING_MCC, &mcc))) {
+            AssertFatal(
+                strlen(mcc) == MAX_MCC_LENGTH,
+                "Bad MCC length (%ld), it must be %u digit ex: 001\n",
+                strlen(mcc), MAX_MCC_LENGTH);
+            char c[2]                                     = {mcc[0], 0};
```

Comment:
can you instead use `{mcc[0], '\0'}` and comment that this is a null terminated string?